apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/CommonTransformerUtils.java

```diff
@@ -0,0 +1,594 @@
+import gov.cms.bfd.model.rif.entities.Beneficiary;
+import gov.cms.bfd.model.rif.entities.CarrierClaim;
+import gov.cms.bfd.server.sharedutils.BfdMDC;
+import gov.cms.bfd.sharedutils.exceptions.BadCodeMonkeyException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import org.hl7.fhir.instance.model.api.IAnyResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Contains shared methods used for looking up NPI and Procedure code descriptions and logging
+ * unfound entries only once.
+ */
+public final class CommonTransformerUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(CommonTransformerUtils.class);
+
+  /** file containing NPI code entries. */
+  private static final String NPI_CODE_FILE = "NPI_Coded_Display_Values_Tab.txt";
+
+  /** file containing NPI code entries. */
+  private static final String PROCEDURE_CODE_FILE = "PRCDR_CD.txt";
+
+  /** Stores the diagnosis ICD codes and their display values. */
+  private static Map<String, String> icdMap = null;
+
+  /** Stores the procedure codes and their display values. */
+  private static Map<String, String> procedureMap = null;
+
+  /** Stores the NPI codes and their display values. */
+  private static Map<String, String> npiMap = null;
+
+  /** default NPI org description for failed code lookup. */
+  private static final String NPI_ORG_DISPLAY_DEFAULT = "UNKNOWN";
+
+  /**
+   * string value used when constructing {@link IdDt} Patient identifier; since the class will never
+   * be renamed, OK to use the STU3 string value.
+   */
+  private static final String PATIENT_SIMPLE_CLASSNAME =
+      org.hl7.fhir.dstu3.model.Patient.class.getSimpleName();
+
+  /**
+   * string value used when constructing {@link IdDt} Coverage identifier; since the class will
+   * never be renamed, OK to use the STU3 string value.
+   */
+  private static final String COVERAGE_SIMPLE_CLASSNAME =
+      org.hl7.fhir.dstu3.model.Coverage.class.getSimpleName();
+
+  /**
+   * Tracks the {@link CcwCodebookInterface} that have already had code lookup failures due to
+   * missing {@link Value} matches. Why track this? To ensure that we don't spam log events for
+   * failed lookups over and over and over. This was needed to fix CBBF-162, where those log events
+   * were flooding our logs and filling up the drive.
+   *
+   * @see TransformerUtils#calculateCodingDisplay(IAnyResource, CcwCodebookInterface, String)
```

Comment:
We should change this comment as this Method is now in the CommonTransformerUtils. Also should we call this class a `TransformerUtilsHelper` since its still supporting the existing TransformerUtils classes. ?

Reply:
fixed. leaving class name as is.